IdentityCore/src/cache/ios/MSIDKeychainTokenCache.m

```diff
@@ -22,30 +22,186 @@
 // THE SOFTWARE.
 
 #import "MSIDKeychainTokenCache.h"
+#import "MSIDTokenCacheKey.h"
+#import "MSIDTokenSerializer.h"
+#import "MSIDKeychainUtil.h"
+#import "MSIDToken.h"
+
+static NSString *const s_libraryString = @"MSOpenTech.ADAL.1";
+static NSString *const s_wipeLibraryString = @"Microsoft.ADAL.WipeAll.1";
+static MSIDKeychainTokenCache *_defaultCache = nil;
 
 @implementation MSIDKeychainTokenCache
 
-- (BOOL)setItemWithKey:(MSIDTokenCacheKey *)key
-                  item:(MSIDToken *)item
-            serializer:(id<MSIDTokenSerializer>)serializer
-               context:(id<MSIDRequestContext>)context
-                 error:(NSError **)error
+#pragma mark - Public
+
++ (NSString *)adalAccessGroup
+{
+    return @"com.microsoft.adalcache";
+}
+
++ (NSString *)appDefaultAccessGroup
+{
+    return MSIDKeychainUtil.appDefaultAccessGroup;
+}
+
++ (MSIDKeychainTokenCache *)defaultKeychainCache
 {
-    return NO;
+    if (!_defaultCache)
+    {
+        _defaultCache = [[MSIDKeychainTokenCache alloc] initWithGroup:MSIDKeychainTokenCache.adalAccessGroup];
+    }
+    
+    return _defaultCache;
+}
+
++ (void)setDefaultKeychainCache:(MSIDKeychainTokenCache *)defaultKeychainCache
+{
+    if (!defaultKeychainCache)
+    {
+        return;
+    }
+    
+    _defaultCache = defaultKeychainCache;
+}
+
+- (instancetype)initWithGroup:(NSString *)accessGroup
+{
+    if (!(self = [super init]))
+    {
+        return nil;
+    }
+    
+    if (!accessGroup)
+    {
+        MSID_LOG_INFO(nil, @"Keychain initialization with nil accessGroup.");
+        return nil;
+    }
+    
+    MSID_LOG_INFO(nil, @"Using keychain accessGroup: %@", _PII_NULLIFY(accessGroup));
+    MSID_LOG_INFO_PII(nil, @"Using keychain accessGroup: %@", accessGroup);
+    
+    _accessGroup = accessGroup;
```

Comment:
The access group used for queries should be [[NSString alloc] initWithFormat:@"%@.%@", teamId, accessGroup]

Reply:
changed